File: app/models/Team.java
package models;

import play.libs.F.Option;
import play.libs.Json;

public class Team {

	private JsonWebSocket out;

	private final static int NOT_BUZZED = 1000;
	private String name;
	private int score = 0;
	private int buzzOrder = NOT_BUZZED;
	
	public Team(String name, JsonWebSocket out) {
		this.name = name;
		this.out = out;
	}

	public String getName() {
		return name;
	}

	public int getScore() {
		return score;
	}
	
	public int getBuzzOrder() {
		return this.buzzOrder;
	}
	
	public void scored(int delta) {
		this.score += delta;
	}	
	
	public void buzzed(int buzzOrder) {
		this.buzzOrder = buzzOrder;
	}
	
	public void resetBuzzer() {
		this.buzzOrder = NOT_BUZZED;
	}
	
	public boolean haveBuzzed() {
		return this.buzzOrder > 0;
	}
	
	public void notify(Option<Object> obj) {
		if(out != null && !obj.isEmpty()) {
			out.write(Json.toJson(obj.get()));
		}
	}

	public static Team nil() {
		return new Team(null, null);
	}
	
}

File: app/models/Team.kt
package models

class Team(val name: String) {
    private var out: JsonWebSocket? = null
    private val NOT_BUZZED = 1000

    var score = 0
    var buzzOrder = NOT_BUZZED

    var latestResponse: String? = null
    var status: Status = Status.JOINING

    var sound: String = "ping.wav"

    fun scored(delta: Int) {
        this.score += delta
    }

    fun buzzed(buzzOrder: Int) {
        this.buzzOrder = buzzOrder
    }

    fun resetBuzzer() {
        this.buzzOrder = NOT_BUZZED
    }

    fun notify(msg: Event) {
        out?.write(msg.asJson())
    }

    fun bind(msg: JsonWebSocket) {
        this.out = msg
    }

    companion object {
        fun nil(): Team {
            return Team("")
        }
    }

    enum class Status {
        JOINING, LIVE, AWOL, GONE
    }
}